### PR TITLE
remove fedora 23

### PR DESCRIPTION
**Pull request by danmosemsft**
Remove more Fedora 23.
@wtgodbe @gkhanna79 

I added 25 to places where it had no runtime significance, but not elsewhere since I'm  not sure of the status of builds of it.

contributes to https://github.com/dotnet/coreclr/pull/10719

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -327,7 +327,7 @@
     "RuntimeId": {
       "description": "Specifies the OS to build Core_Root for",
       "valueType": "property",
-      "values": [ "debian.8-x64", "fedora.23-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64" ],
+      "values": [ "debian.8-x64", "fedora.24-x64", "fedora.25-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64" ],
```
Comment (gkhanna79) [inline config.json]: We do not support F24 in official builds but only in CI, right @wtgodbe?

File: init-tools.sh [hunk 2]
```diff
@@ -44,11 +44,11 @@ get_current_linux_name() {
         return 0
     elif [ "$(cat /etc/os-release | grep -cim1 fedora)" -eq 1 ]; then
         if [ "$(cat /etc/os-release | grep -cim1 23)" -eq 1 ]; then
-            echo "fedora.23"
+            echo "fedora.24"
```
Comment (gkhanna79) [inline init-tools.sh]: You need to fix the earlier statement as well. Same for the next change.

File: config.json (hunk 1 above)
Comment (danmosemsft) [inline config.json]: @wtgodbe ?

Comment (wtgodbe) [inline config.json]: No, we build Fedora 24 in the pipeline as well.

**Comment by danmosemsft** [comment]
@gkhanna79 can you sign off?

**Comment by ahsonkhan** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please (network again)

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please (hope springs eternal)

**Comment by ahsonkhan** [comment]
@danmosemsft, why does this keep happening? Is there a way to resolve it?

**Comment by gkhanna79** [comment]
Merging this PR since OSX infra is down and out.

**Pull request merged**